### PR TITLE
Make CLI run file metrics

### PR DESCRIPTION
**Pull request by xplosunn**

### CODE REVIEW

File: cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala [hunk 1]
```diff
@@ -155,13 +180,31 @@ class AnalyseExecutor(toolInput: Option[String],
 
 object AnalyseExecutor {
 
-  final case class ExecutorResult(toolName: String, files: Set[Path], analysisResults: Try[Set[ToolResult]])
+  final case class ExecutorResult(toolName: String, files: Set[Path], analysisResults: Try[Set[Result]])
+
+  def languages(localConfiguration: Either[String, CodacyConfigurationFile],
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala]: Why do you need this method? Seems like a strait invocation. If you want to optimise I would do it inside.

File: cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala [hunk 2]
```diff
@@ -49,21 +51,32 @@ class AnalyseExecutor(toolInput: Option[String],
...
 
     val analysisResult: Either[String, Seq[ExecutorResult]] = filesTargetAndTool.map {
       case (filesTarget, tools) =>
-        SetOps.mapInParallel(tools, nrParallelTools) { tool: Tool =>
-          val analysisResults: Try[Set[ToolResult]] = analyseFiles(tool, filesTarget, localConfigurationFile)
-
+        SetOps.mapInParallel(tools, nrParallelTools) { tool: ITool =>
+          val analysisResults: Try[Set[Result]] = tool match {
+            case tool: Tool =>
+              analyseFiles(tool, filesTarget, localConfigurationFile).map(_.map(_.to[Result]))
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala]: better name -> `tools` | `issues` | `...`

File: cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala [hunk 3]
```diff
@@ -75,7 +88,9 @@ class AnalyseExecutor(toolInput: Option[String],
   private def analyseFiles(tool: Tool,
                            filesTarget: FilesTarget,
                            localConfigurationFile: Either[String, CodacyConfigurationFile]): Try[Set[ToolResult]] = {
-    val fileTarget = fileCollector.filter(tool, filesTarget, localConfigurationFile, remoteProjectConfiguration)
+    val analysisFilesTarget =
+      fileCollector.filter(tool, filesTarget, localConfigurationFile, remoteProjectConfiguration)
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala]: We need to filter files here by the language we are running and not the languages the tool supports

Comment (xplosunn) [reply cli/src/main/scala/com/codacy/analysis/cli/command/analyse/AnalyseExecutor.scala]: I think I want to do this in a different PR.

**Pull request merged**